web/src/compose_actions.js

```diff
@@ -290,6 +290,8 @@ export function start(msg_type, opts) {
 
     // Show a warning if topic is resolved
     compose_validate.warn_if_topic_resolved(true);
+    // Show a warning if the user is in a search narrow
+    compose_validate.warn_if_in_search_view();
```

Comment:
Hmm, should this be conditional in some way on how you opened the compose box? I feel like if I hit `X` to start composing a new DM, while I happen to be looking at my mentions view, this banner is not appropriate.

Do we want it to just apply when we're a reply?

Reply:
Ooh, yeah, that probably makes more sense -- a reply via the keyboard shortcut or the big "Message..." button.

Reply:
That makes a lot of sense! Have changed the code to only show the banner when replying to a message in a search view.